server/recovery.go

```diff
@@ -6,16 +6,30 @@
 
 package server
 
-import "net/http"
+import (
+	"net/http"
+)
 
 func Recovery(h http.Handler) http.Handler {
 	return http.HandlerFunc(func(rw http.ResponseWriter, r *http.Request) {
+		var statusCode *int
+
 		defer func() {
-			if err := recover(); err != nil {
-				rw.WriteHeader(http.StatusInternalServerError)
-				panic(err)
+			if statusCode != nil {
+				rw.WriteHeader(*statusCode)
 			}
 		}()
+
+		defer RecoverAndSetStatusCode(&statusCode)
+
 		h.ServeHTTP(rw, r)
 	})
 }
+
+func RecoverAndSetStatusCode(statusCode **int) {
```

Comment:
this does not need to be a `**int`, just an `*int`.

Reply:
I want to change this value https://github.com/theplant/appkit/pull/32/files#diff-7cb21773652077f716d8056ef91f6c35R15, if we use `**int` then we can change value of `*int` (pointer), if pointer value is not nil then mean we have set status code.

If we use `*int`, then the logic is

```go
if statusCode != 0 {
	rw.WriteHeader(*statusCode)
}
```

I think `!= nil` is better than `!= 0` to judge whether a value has been set, because `0` has two meanings, one is default value, one is we set to `0`.

Reply:
`0` is not a valid HTTP status code ;)

Reply:
haha, I changed from `**int` to `*int` in https://github.com/theplant/appkit/pull/32/commits/52d181dd491e189670b1c075dec18a903ac957d9